registry/storage/tagstore.go

```diff
@@ -2,28 +2,46 @@ package storage
 
 import (
 	"context"
+	"os"
 	"path"
 	"sort"
+	"strconv"
+	"sync"
 
 	"github.com/distribution/distribution/v3"
 	storagedriver "github.com/distribution/distribution/v3/registry/storage/driver"
 	"github.com/opencontainers/go-digest"
+	"github.com/sirupsen/logrus"
 )
 
-var _ distribution.TagService = &tagStore{}
+var _ distribution.TagService = &TagStore{}
 
-// tagStore provides methods to manage manifest tags in a backend storage driver.
+// TagStore provides methods to manage manifest tags in a backend storage driver.
 // This implementation uses the same on-disk layout as the (now deleted) tag
 // store.  This provides backward compatibility with current registry deployments
 // which only makes use of the Digest field of the returned distribution.Descriptor
 // but does not enable full roundtripping of Descriptor objects
-type tagStore struct {
-	repository *repository
-	blobStore  *blobStore
+type TagStore struct {
+	repository              *repository
+	blobStore               *blobStore
+	lookupConcurrencyFactor int
+}
+
+func NewStore(repository *repository, blobStore *blobStore) *TagStore {
+	lookupConcurrencyFactor, err := strconv.Atoi(os.Getenv("STORAGE_TAGSTORE_LOOKUP_CONCURRENCY"))
+	if err != nil {
+		lookupConcurrencyFactor = 64
+		logrus.Infof("TagStore: STORAGE_TAGSTORE_LOOKUP_CONCURRENCY is not set. Using default %d as lookup concurrency factor", lookupConcurrencyFactor)
```

Comment:
See my earlier comment about passing down `ctx` to `NewStore` and extracting the logger from it. There is no need to make the `ctx` a field of `TagStore`